lib/SILOptimizer/Utils/CanonicalizeBorrowScope.cpp

```diff
@@ -261,8 +263,13 @@ bool CanonicalizeBorrowScope::visitBorrowScopeUses(SILValue innerValue,
         // Pointer escapes are only allowed if they use the guaranteed value,
         // which means that the escaped value must be confied to the current
         // borrow scope.
-        if (use->get().getOwnershipKind() != OwnershipKind::Guaranteed)
+        if (use->get().getOwnershipKind() != OwnershipKind::Guaranteed
+            && !isa<SILFunctionArgument>(innerValue)) {
+          return false;
+        }
+        if (!visitor.visitUse(use)) {
```

Comment:
Would asserting inside the condition that `!isa<SILFunctionArgument>(innerValue)` be worthwhile?